File: client/src/app/core/services/work-item-link.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { API_BASE_URL } from '../config/api.config';
import { WorkItemLinkCreateRequest, WorkItemLinkResponse } from '../models/work-item-link.model';

const BASE = `${API_BASE_URL}/api`;

@Injectable({ providedIn: 'root' })
export class WorkItemLinkService {
  private readonly http = inject(HttpClient);

  list(workItemId: string): Observable<WorkItemLinkResponse[]> {
    return this.http.get<WorkItemLinkResponse[]>(`${BASE}/work-items/${workItemId}/links`);
  }

  create(workItemId: string, request: WorkItemLinkCreateRequest): Observable<WorkItemLinkResponse> {
    return this.http.post<WorkItemLinkResponse>(`${BASE}/work-items/${workItemId}/links`, request);
  }

  delete(workItemId: string, linkId: string): Observable<void> {
    return this.http.delete<void>(`${BASE}/work-items/${workItemId}/links/${linkId}`);
  }
}

File: Customer/src/main/java/com/postman/alt/service/WorkItemLinkService.java
package com.postman.alt.service;

import com.postman.alt.service.dto.WorkItemLinkCreateRequest;
import com.postman.alt.service.dto.WorkItemLinkResponse;

import java.util.List;
import java.util.UUID;

public interface WorkItemLinkService {

    // both outgoing (this item is the source) and incoming links.
    List<WorkItemLinkResponse> list(UUID workItemId, UUID requesterId);

    WorkItemLinkResponse create(UUID workItemId, UUID requesterId, WorkItemLinkCreateRequest request);

    void delete(UUID workItemId, UUID linkId, UUID requesterId);
}
